firedrake/utility_meshes.py

```diff
@@ -863,6 +864,118 @@ def UnitIcosahedralSphereMesh(refinement_level=0, degree=1, reorder=None,
                                  comm=comm)
 
 
+def OctahedralSphereMesh(radius, refinement_level=0, degree=1,
+                         hemisphere="both", reorder=None,
+                         comm=COMM_WORLD):
+    """Generate an octahedral approximation to the surface of the
+    sphere.
+
+    :arg radius: The radius of the sphere to approximate.
+    :kwarg refinement_level: optional number of refinements (0 is an
+        octahedron).
+    :kwarg degree: polynomial degree of coordinate space (defaults
+        to 1: flat triangles)
+    :kwarg hemisphere: One of "both" (default), "north", or "south"
```

Comment:
I prefer that one can ask for "both", "south" or "north" for the hemisphere argument like this.